core/src/main/java/cz/xtf/core/openshift/OpenShift.java

```diff
@@ -292,21 +297,88 @@ public String getPodLog(String deploymentConfigName) {
 	}
 
 	public String getPodLog(Pod pod) {
-		return pods().withName(pod.getMetadata().getName()).getLog();
+		return getPodLog(pod, getAnyContainer(pod));
+	}
+	
+	public String getPodLog(Pod pod, String containerName) {
+		return getPodLog(pod, getContainerByName(pod, containerName));
+	}
+
+	public String getPodLog(Pod pod, Container container) {
+		if (Objects.nonNull(container)) {
+			return pods().withName(pod.getMetadata().getName()).inContainer(container.getName()).getLog();
+		} else {
+			return pods().withName(pod.getMetadata().getName()).getLog();
+		}
+	}
+
+	/**
+	 * Return logs of all containers from the pod
+	 * 
+	 * @param pod Pod to retrieve from
+	 * @return Map of container name / logs
+	 */
+	public Map<String, String> getAllContainerLogs(Pod pod) {
+		return retrieveFromPodContainer(pod, container -> this.getPodLog(pod, container));
 	}
 
 	public Reader getPodLogReader(Pod pod) {
-		return pods().withName(pod.getMetadata().getName()).getLogReader();
+		return getPodLogReader(pod, getAnyContainer(pod));
+	}
+
+	public Reader getPodLogReader(Pod pod, Container container) {
```

Comment:
Please reorganize the method so the final its top down from the one with the simplest parameters to the actual implementation:
```
getPodLogReader(Pod pod)
getPodLogReader(Pod pod, String containerName)
getPodLogReader(Pod pod, Container container)
```

Applies also for others updated and created methods.

Reply:
updated